substrate/frame/aura/src/lib.rs

```diff
@@ -311,6 +309,48 @@ impl<T: Config> Pallet<T> {
 
 		Ok(())
 	}
+
+	/// Returns the current set of Aura authorities.
+	///
+	/// This function retrieves the current set of authorities responsible for producing blocks in
+	/// the Aura consensus mechanism. The authorities are returned as a `BoundedVec` to ensure the
+	/// collection does not exceed a predefined maximum size, which is specified by
+	/// the `MaxAuthorities` trait bound.
+	///
+	/// # Returns
+	///
+	/// A `BoundedVec` containing the current set of Aura authorities. The type of each authority
+	/// is defined by `T::AuthorityId`, which is associated with the specific implementation of the
+	/// Aura consensus in the pallet.
+	///
+	/// # Deprecation note
+	///
+	/// This function will be deprecated in future releases. It is recommended not to use it in new
+	/// developments.
+	pub fn authorities() -> BoundedVec<T::AuthorityId, T::MaxAuthorities> {
```

Comment:
where are these functions used? do they even need to exist?

Reply:
Thank you for your review and the points raised. I appreciate the opportunity to clarify the usage and implications of the proposed changes.

I've noted that the this function is utilized not only within this pallet like [here](https://github.com/paritytech/polkadot-sdk/pull/3350/files/f92d601fbe50e45bced1ede2181dc81edd3013a1#diff-be71be33bddf8c17afa2bac0568bd30991b1098634bb3cd8a101545052849cd7R378) but also across other crates within the SDK like[ here](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachain-template/runtime/src/lib.rs#L549). This widespread use underscores its current necessity and potential impact on existing implementations.

In my understanding, the primary aim of this PR is to streamline the codebase by removing unnecessary macro complexity, thereby improving readability and maintainability without altering the existing functionality or interfaces. Given this scope, maintaining these function calls as-is ensures compatibility and stability across the SDK during this transitional phase. Please suggest .

If recommended I can go ahead and replace the uses here with Authorities::<T>::get() in a future iteration.

Let me know if you recommend proceeding with these adjustments now, or if we should consider them as part of a subsequent phase of optimization.

Reply:
>If recommended I can go ahead and replace the uses here with Authorities::<T>::get() in a future iteration.

In my opinion we should do this. No need for another function.

Reply:
Thank you for your valuable feedback. I've implemented the suggested changes in the latest commit (e81e6c4). You can view the changes[ here](https://github.com/paritytech/polkadot-sdk/pull/3350/commits/e81e6c4f86c06fd0047c6b333ff5a59e438a32c3).

Please take a look at the updated code when you have a moment. I'm open to any further suggestions or discussions if there's more to refine.

Reply:
@k-gunjan I think Oliver meant that you should not introduce these new methods in this PR.
Instead, you should go around the other crates that call the getter functions you removed, and replace them with `pallet_aura::Authorities::<T>::get()`.

Reply:
Thanks Muraca ! I'll remove definition from here and replace uses in other pallets.